libp2p/protocols/kademlia/provider.nim

```diff
@@ -39,20 +80,48 @@ proc addProvider*(kad: KadDHT, cid: Cid) {.async: (raises: [CancelledError]), gc
       # Dispatch will timeout if any of the calls don't receive a response (which is normal)
       discard
 
+proc startProviding*(kad: KadDHT, c: Cid) {.async: (raises: [CancelledError]).} =
```

Comment:
Something to consider:
you might want to provide a CID for some Duration, or even provide it infinitely (, thus maybe you'd want to have a signature like:
```nim
proc startProviding*(kad: KadDHT, c: Cid, duration: InfiniteDuration)
```
would require `providedKeys` to be something like:
```nim
type ProvidedContent = object
   cid: Cid
   expiresAt*: chronos.Moment

type ProviderManager* = ref object
  ...
  providedKeys*: HashSet[ProvidedContent]
```
and the republish to check if the provided content isnt expired or GC it otherwise